# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/12/27 20:55:17-08:00 matthew@wil.cx 
#   [SPARC64]: Stop referencing i_sock directly.
#   
#   Use S_ISSOCK() instead.
#   
#   Signed-off-by: Matthew Wilcox <matthew@wil.cx>
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
# arch/sparc64/solaris/ioctl.c
#   2004/12/27 20:54:56-08:00 matthew@wil.cx +2 -2
#   [SPARC64]: Stop referencing i_sock directly.
#   
#   Use S_ISSOCK() instead.
#   
#   Signed-off-by: Matthew Wilcox <matthew@wil.cx>
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
# arch/sparc64/solaris/socksys.c
#   2004/12/27 20:54:56-08:00 matthew@wil.cx +1 -1
#   [SPARC64]: Stop referencing i_sock directly.
#   
#   Use S_ISSOCK() instead.
#   
#   Signed-off-by: Matthew Wilcox <matthew@wil.cx>
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
# arch/sparc64/solaris/timod.c
#   2004/12/27 20:54:56-08:00 matthew@wil.cx +2 -4
#   [SPARC64]: Stop referencing i_sock directly.
#   
#   Use S_ISSOCK() instead.
#   
#   Signed-off-by: Matthew Wilcox <matthew@wil.cx>
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
diff -Nru a/arch/sparc64/solaris/ioctl.c b/arch/sparc64/solaris/ioctl.c
--- a/arch/sparc64/solaris/ioctl.c	2005-01-04 17:50:57 -08:00
+++ b/arch/sparc64/solaris/ioctl.c	2005-01-04 17:50:57 -08:00
@@ -298,7 +298,7 @@
 	if (! current->files->fd[fd] ||
 	    ! current->files->fd[fd]->f_dentry ||
 	    ! (ino = current->files->fd[fd]->f_dentry->d_inode) ||
-	    ! ino->i_sock) {
+	    ! S_ISSOCK(ino->i_mode)) {
 		spin_unlock(&current->files->file_lock);
 		return TBADF;
 	}
@@ -478,7 +478,7 @@
         struct module_info *mi;
 
         ino = filp->f_dentry->d_inode;
-        if (! ino->i_sock)
+        if (!S_ISSOCK(ino->i_mode))
 		return -EBADF;
         sock = filp->private_data;
         if (! sock) {
diff -Nru a/arch/sparc64/solaris/socksys.c b/arch/sparc64/solaris/socksys.c
--- a/arch/sparc64/solaris/socksys.c	2005-01-04 17:50:57 -08:00
+++ b/arch/sparc64/solaris/socksys.c	2005-01-04 17:50:57 -08:00
@@ -150,7 +150,7 @@
 	unsigned int mask = 0;
 
 	ino=filp->f_dentry->d_inode;
-	if (ino && ino->i_sock) {
+	if (ino && S_ISSOCK(ino->i_mode)) {
 		struct sol_socket_struct *sock;
 		sock = (struct sol_socket_struct*)filp->private_data;
 		if (sock && sock->pfirst) {
diff -Nru a/arch/sparc64/solaris/timod.c b/arch/sparc64/solaris/timod.c
--- a/arch/sparc64/solaris/timod.c	2005-01-04 17:50:57 -08:00
+++ b/arch/sparc64/solaris/timod.c	2005-01-04 17:50:57 -08:00
@@ -853,9 +853,7 @@
 	if(!filp) goto out;
 
 	ino = filp->f_dentry->d_inode;
-	if (!ino) goto out;
-
-	if (!ino->i_sock)
+	if (!ino || !S_ISSOCK(ino->i_mode))
 		goto out;
 
 	ctlptr = (struct strbuf __user *)A(arg1);
@@ -923,7 +921,7 @@
 	ino = filp->f_dentry->d_inode;
 	if (!ino) goto out;
 
-	if (!ino->i_sock &&
+	if (!IS_SOCK(ino->i_mode) &&
 		(imajor(ino) != 30 || iminor(ino) != 1))
 		goto out;
 
